src/Http/Http.Results/src/HttpResultsWriter.cs

```diff
@@ -142,6 +153,12 @@ public static void ApplyProblemDetailsDefaultsIfNeeded(object? value, int? statu
         }
     }
 
+    private static JsonOptions ResolveJsonOptions(HttpContext httpContext)
+    {
+        // Attempt to resolve options from DI then fallback to default options
+        return _defaultOptions ??= httpContext.RequestServices?.GetService<IOptions<JsonOptions>>()?.Value ?? new JsonOptions();
```

Comment:
We shouldn't be caching here

Reply:
👍, any specific reasons why not?

Reply:
Caches need to be associated with the service provider. Which adds unnecessary complexity to this code path.